app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetController.scala

```diff
@@ -145,11 +163,48 @@ class EditSetController @Inject() (
   def view(id: String): Action[AnyContent] = Action { implicit request: Request[AnyContent] =>
     withUser { user =>
       logger.info(s"The edit set id is $id ")
-      val editSetModels = editSets.getEditSet()
-      val messages: Messages = request.messages
-      val title: String = messages("edit-set.title")
-      val heading: String = messages("edit-set.heading", editSetModels.name)
-      Ok(editSet(user, title, heading, editSetModels))
+      generateEditSetView(user, fallbackEditSetReorder)
+    }
+  }
+
+  def viewAfterReordering(idCurrentlyUnused: String): Action[AnyContent] = Action {
+    implicit request: Request[AnyContent] =>
+      withUser { user =>
+        generateEditSetView(user, formToEither(reorderForm.bindFromRequest()).getOrElse(fallbackEditSetReorder))
+      }
+  }
+
+  private def generateEditSetView(user: User, editSetReorder: EditSetReorder)(implicit
+    request: Request[AnyContent]
+  ): Result = {
+    val currentEditSet = editSets.getEditSet()
+    val title = resolvedMessage("edit-set.title")
+    val heading: String = resolvedMessage("edit-set.heading", currentEditSet.name)
+    val editSetEntries = currentEditSet.entries.sortWith(getSorter(editSetReorder))
+    Ok(editSet(user, title, heading, editSetEntries, reorderForm.fill(editSetReorder)))
+  }
+
+  private def getSorter(editSetReorder: EditSetReorder): (EditSetEntry, EditSetEntry) => Boolean = {
```

Comment:
Rather than using `sortWith` and a function from `(EditSetEntry, EditSetEntry) => Boolean` the `Ordering` type class could be used to cut down on function definitions:
```
    private def getSorter(editSetReorder: EditSetReorder): Ordering[EditSetEntry] = {
      val fieldOrdering: Ordering[EditSetEntry] = editSetReorder.field match {
        case FieldNames.ccr             => Ordering.by(_.ccr)
        case FieldNames.scopeAndContent => Ordering.by(_.scopeAndContent)
        case FieldNames.coveringDates   => Ordering.by(_.coveringDates)
        case _                          => Ordering.by(_.ccr)
      }

      if (editSetReorder.direction == orderDirectionDescending) fieldOrdering.reverse else fieldOrdering
    }
```
and 
```
val editSetEntries = currentEditSet.entries.sortWith(getSorter(editSetReorder))
```
would be 
```
val editSetEntries = currentEditSet.entries.sorted(getSorter(editSetReorder))
```
The advantage I see would be to reduce the number of changes needed when a new sort method is used.

Reply:
Good idea. I prefer it. Will change.

Reply:
Changed.